lib/ansible/module_utils/network/nxos/config/telemetry/telemetry.py

```diff
@@ -0,0 +1,287 @@
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_tms_global_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        config = self._module.params['config']
+        want = dict((k, v) for k, v in config.items() if v is not None)
+        have = existing_tms_global_facts
+        resp = self.set_state(want, have)
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        # Compare want and have states first.  If equal then return.
+        state = self._module.params['state']
+        if 'overridden' in state or 'replaced' in state:
+            self._module.fail_json(msg='State: <{0}> not yet supported'.format(state))
+
+        # Save off module params
+        ALL_MP = self._module.params['config']
+
+        cmd_ref = {}
+        cmd_ref['TMS_GLOBAL'] = {}
+        cmd_ref['TMS_DESTGROUP'] = {}
+        cmd_ref['TMS_SENSORGROUP'] = {}
+        cmd_ref['TMS_SUBSCRIPTION'] = {}
+
+        # Get Telemetry Global Data
+        cmd_ref['TMS_GLOBAL']['ref'] = []
+        self._module.params['config'] = get_module_params_subsection(ALL_MP, 'TMS_GLOBAL')
+        cmd_ref['TMS_GLOBAL']['ref'].append(NxosCmdRef(self._module, TMS_GLOBAL))
+        ref = cmd_ref['TMS_GLOBAL']['ref'][0]
+        ref.set_context()
+        ref.get_existing()
+        ref.get_playvals()
+        device_cache = ref.cache_existing
+
+        if device_cache is None:
+            device_cache_lines = []
+        else:
+            device_cache_lines = device_cache.split("\n")
+
+        # Get Telemetry Destination Group Data
+        if want.get('destination_groups'):
+            td = {'name': 'destination_groups', 'type': 'TMS_DESTGROUP',
+                  'obj': TMS_DESTGROUP, 'cmd': 'destination-group {0}'}
+            cmd_ref[td['type']]['ref'] = []
+            saved_ids = []
+            for playvals in want[td['name']]:
+                valiate_input(playvals, td['name'], self._module)
+                if playvals['id'] in saved_ids:
+                    continue
+                saved_ids.append(playvals['id'])
+                resource_key = td['cmd'].format(playvals['id'])
+                self._module.params['config'] = get_module_params_subsection(ALL_MP, td['type'], playvals['id'])
+                cmd_ref[td['type']]['ref'].append(NxosCmdRef(self._module, td['obj']))
```

Comment:
What's happening with the two lines above? Can you add some comments to the code regarding intent?